A2A Customer Query Routing and Resolution/multi_agent/agent.py

```diff
@@ -0,0 +1,570 @@
+import requests
+from a2a.server.agent_execution import AgentExecutor, RequestContext
+from a2a.server.apps import A2AStarletteApplication
+from a2a.server.events import EventQueue
+from a2a.server.request_handlers import DefaultRequestHandler
+from a2a.server.tasks import InMemoryTaskStore, TaskUpdater
+from a2a.types import (
+    AgentCapabilities,
+    AgentCard,
+    AgentSkill,
+    MessageSendParams,
+    Part,
+    TaskState,
+    TextPart,
+    SendMessageRequest,
+)
+from a2a.utils import new_agent_text_message, new_task
+from a2a.client import A2AClient
+
+# LlamaIndex imports for knowledge base functionality
+from llama_index.core import Document, VectorStoreIndex
+from llama_index.core.node_parser import SentenceSplitter
+from llama_index.embeddings.litellm import LiteLLMEmbedding
+from llama_index.llms.litellm import LiteLLM as LlamaIndexLiteLLM_LLM
+
+# Google ADK imports for agent creation
+from google.adk.agents import LlmAgent
+from google.adk.tools.agent_tool import AgentTool
+from google.adk.runners import Runner
+from google.adk.sessions import InMemorySessionService
+from google.adk.memory.in_memory_memory_service import InMemoryMemoryService
+from google.adk.artifacts import InMemoryArtifactService
+from google.adk.models.lite_llm import LiteLlm
+from litellm import completion
+from google.genai import types
+
+# Setup logging for debugging and monitoring
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+load_dotenv()
+
+# Knowledge Base class for handling customer support queries
+class KB:
+    def __init__(self):
+        # Load knowledge base from JSON file
+        try:
+            path = Path("knowledge_base/swiftcart_kb.json")
+            data = json.loads(path.read_text())
+        except FileNotFoundError:
+            logger.error("knowledge_base/swiftcart_kb.json not found. Please create it.")
+            data = {}
+
+        # Convert JSON data to LlamaIndex documents
+        docs = [
+            Document(
+                text=f"Q: {faq['question']}\nA: {faq['answer']}",
+                metadata={"category": cat}
+            )
+            for cat, faqs in data.items()
+            for faq in faqs
+            if isinstance(faq, dict) and 'question' in faq and 'answer' in faq
+        ]
+
+        # Add placeholder if no documents found
+        if not docs:
+            logger.warning("No documents loaded into Knowledge Base. Queries will likely fail.")
+            docs.append(Document(text="Placeholder document."))
+
+        # Split documents into chunks for better retrieval
+        nodes = SentenceSplitter(chunk_size=512, chunk_overlap=20).get_nodes_from_documents(docs)
+
+        # Get API credentials for LLM services
+        api_base = os.getenv("NEBIUS_API_BASE")
+        api_key = os.getenv("NEBIUS_API_KEY")
+
+        # Create vector index for semantic search
+        self.index = VectorStoreIndex(
+            nodes,
+            embed_model=LiteLLMEmbedding(
```

Comment:
Please use direct [Nebius/LiteLLM](https://docs.litellm.ai/docs/providers/nebius) model here instead of Open AI method.

Same at other places as well!